controllers/dishes.js

```diff
@@ -0,0 +1,48 @@
+import Dishes from "../models/dishes.js";
+
+export async function getAllDishes(req, res, next) {
+  try {
+    const { page = 1, limit = 10, q = "" } = req.query;
+    const filter = { name: { $regex: q, $options: "i" } };
+    const findPromise = Dishes.find(filter, {}, { skip: limit * (page - 1), limit })
+      .populate("ingredients.ingredient")
+      .populate("createdBy", "name fName lName")
+      .populate("updatedBy", "name fName lName");
+    const countPromise = Dishes.countDocuments(filter);
+    const [dishes, count] = await Promise.all([findPromise, countPromise]);
+    res.json({ data: dishes, count });
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function createDish(req, res, next) {
+  try {
+    const { body } = req;
+    // TODO: Update createdBy and updatedBy
+    body.createdBy = "Admin";
+    body.updatedBy = "Admin";
```

Comment:
@Van-sh These will be set to current user now.

Reply:
I updated them